### PR TITLE
feat(wit): workflow for wasmcloud:bus

### PR DESCRIPTION
**Pull request by ricochet**
Signed-off-by: Bailey Hayes <behayes2@gmail.com>

### CODE REVIEW

**Review by ricochet — COMMENTED** [review]
For those reviewing, note that after this is merged, we need to actually push the appropriate tag wit-wasmcloud-bus-v0.1.0 for the release workflow.

File: .github/workflows/wit-wasmcloud-lattice-bus.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    - name: Extract tag context
+      id: ctx
+      run: |
+          version=${GITHUB_REF_NAME#wit-wasmcloud-secrets-v}
+          echo "version=${version}" >> "$GITHUB_OUTPUT"
+          echo "tarball=wit-wasmcloud-secrets-${version}.tar.gz" >> "$GITHUB_OUTPUT"
+          echo "version is ${version}"
+    - name: Build
+      run: |
+        tar cvzf ${{ steps.ctx.outputs.tarball }} -C wit secrets/wit
```
Comment (joonas) [inline .github/workflows/wit-wasmcloud-lattice-bus.yml:31]:
```suggestion
        tar -cvzf ${{ steps.ctx.outputs.tarball }} -C wit secrets/wit
```

Comment (ricochet) [reply .github/workflows/wit-wasmcloud-lattice-bus.yml]: Thanks! Caught a real bug with my copy-paste. Weirdly tar is a-ok with not passing in a hyphen here. I adjusted the other workflows I copied from to include this as well.

File: crates/component/src/lib.rs [hunk 2]
```diff
@@ -28,7 +28,10 @@ pub mod wasi {
     pub use super::bindings::wasi::*;
     pub use ::wasi::*;
 }
-pub use bindings::wasmcloud;
+
+pub mod wasmcloud {
+    pub use super::bindings::wasmcloud::*;
+}
```
Comment (rvolosatovs) [inline crates/component/src/lib.rs:34]: I'm a bit confused here, doesn't this achieve the same result as above?

File: .github/workflows/wit-wasmcloud-lattice-bus.yml (hunk 1 above)
Comment (rvolosatovs) [reply .github/workflows/wit-wasmcloud-lattice-bus.yml]: There's no bug here. `-` is not required by either GNU nor BSD tar (i.e. neither on Linux, nor on Mac by default), "hyphen-less" `tar` usage is actually much more common

File: crates/component/src/lib.rs (hunk 2 above)
Comment (ricochet) [reply crates/component/src/lib.rs:34]: By moving the bus wit down a level, it changed the bindings generation. I liked keeping the bindings in this file the same with wasmcloud::bus vs bus

**Review by rvolosatovs — APPROVED** [review]

**Pull request merged**